lib/Mongo/MongoCollection.php

```diff
@@ -434,7 +454,18 @@ public function count($query = array())
      * @return array|boolean If w was set, returns an array containing the status of the save.
      * Otherwise, returns a boolean representing if the array was not empty (an empty array will not be inserted).
      */
-    public function save($a, array $options = array()) {}
+    public function save($a, array $options = array())
+    {
+        if (is_object($a)) {
+            $a = (array)$a;
+        }
+        if ( ! array_key_exists('_id', $a)) {
+            $a['_id'] = new \MongoId();
+        }
+        $filter = ['_id' => $a['_id']];
+        // TODO: not sure how to generate result array
```

Comment:
I haven't looked at the return arrays just yet - we can leave this for later as well.